graphyte.py

```diff
@@ -64,31 +64,39 @@ def stop(self):
             self._thread.join()
             self.interval = None
 
-    def build_message(self, metric, value, timestamp):
+    def build_message(self, metric, value, timestamp, tags={}):
         """Build a Graphite message to send and return it as a byte string."""
         if not metric or metric.split(None, 1)[0] != metric:
             raise ValueError('"metric" must not have whitespace in it')
         if not isinstance(value, (int, float)):
             raise TypeError('"value" must be an int or a float, not a {}'.format(
-                    type(value).__name__))
+                type(value).__name__))
+        if not isinstance(tags, dict):
```

Comment:
I'd prefer to remove this `isinstance()` check entirely -- in the spirit of duck typing, let's just allow any dict-like object (in this case anything that supports `.items()`).